### ISSUE TITLE
No `swiftSettings:` for `.plugin` targets?

### ISSUE THREAD

**Issue by dabrahams** [apple/swift-package-manager#6600]
### Description

There's apparently no way to specify, e.g. `-warnings-as-errors` for a `.plugin` target.  Seems like an obvious oversight.

### Swift Package Manager version/commit hash

Swift Package Manager - Swift 5.8.0

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.75.2 Apple Swift version 5.8 (swiftlang-5.8.0.124.2 clang-1403.0.22.11.100)
Target: arm64-apple-macosx13.0
Darwin DaveA-MBP14.localdomain 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000 arm64

**Comment by tomerd** [comment]
cc @neonichu @abertelrud

**Comment by neonichu** [comment]
Yes, that is correct. Not sure if it was an oversight, it matches the proposal.

**Comment by dabrahams** [comment]
It can be kludged around by putting all the functionality in a library, so there's no good reason to disallow the settings directly in the plugin.

**Comment by tomerd** [comment]
@abertelrud do you remember the reasoning for not exposing it?
